homeassistant/components/hue/sensor_device.py

```diff
@@ -0,0 +1,53 @@
+"""Support for the Philips Hue sensor devices."""
+from .const import DOMAIN as HUE_DOMAIN
+
+
+class GenericHueDevice:
+    """Representation of a Hue device."""
+
+    def __init__(self, sensor, name, bridge, primary_sensor=None):
+        """Initialize the sensor."""
+        self.sensor = sensor
+        self._name = name
+        self._primary_sensor = primary_sensor
+        self.bridge = bridge
+
+    @property
+    def primary_sensor(self):
+        """Return the primary sensor entity of the physical device."""
+        return self._primary_sensor or self.sensor
+
+    @property
+    def device_id(self):
+        """Return the ID of the physical device this sensor is part of."""
+        return self.unique_id[:23]
+
+    @property
+    def unique_id(self):
+        """Return the ID of this Hue sensor."""
+        return self.sensor.uniqueid
+
+    @property
+    def name(self):
+        """Return a friendly name for the sensor."""
+        return self._name
+
+    @property
+    def swupdatestate(self):
+        """Return detail of available software updates for this device."""
```

Comment:
@balloob, this `swupdatestate` property was defined but unused in sensors. Should this be included in the `device_info` attributes?

Reply:
No. It's do indicate if a software update is available. I think that we check it for lights and notify user if there is one.

Reply:
> I think that we check it for lights and notify user if there is one.

That's right, it's the only usage, in https://github.com/home-assistant/core/blob/dd1608db0dfe0d140f9c5e92f57e97df5d87e46d/homeassistant/components/hue/light.py#L220

But there it is accessing the property from the `aiohue` object. 

This property is not called anywhere, that's why I was asking about